docs: add a user doc for solutions deployment

alexandre-allard opened the pull request:
**Component**:

<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->

**Context**: 

**Summary**:

**Acceptance criteria**: 


---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #ISSUE_NUMBER

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

alexandre-allard committed:
docs: add a user doc for solutions deployment

Refs: #2309

bert-e commented:
# Hello alexandre-allard-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `documentation/2309-enhance-solutions-doc` into
`development/2.4`
* [w/2.5/documentation/2309-enhance-solutions-doc](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/documentation/2309-enhance-solutions-doc) will be merged into `development/2.5` 
* [w/2.6/documentation/2309-enhance-solutions-doc](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/documentation/2309-enhance-solutions-doc) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

gdemonet commented on docs/operation/solutions.rst:
```diff
@@ -0,0 +1,50 @@
+Solutions Guide
+===============
+
+To deploy a Solution in a MetalK8s cluster, an utility script is provided.
```
```suggestion
To deploy a Solution in a MetalK8s cluster, a utility script is provided.
```

alexandre-allard replied:
done

gdemonet commented on docs/operation/solutions.rst:
```diff
@@ -0,0 +1,50 @@
+===============
+
+To deploy a Solution in a MetalK8s cluster, an utility script is provided.
+This section describe, step by step, how to deploy a Solution using this
```
```suggestion
This section describes, step by step, how to deploy a Solution using this
```

alexandre-allard replied:
done

gdemonet commented on docs/operation/solutions.rst:
```diff
@@ -0,0 +1,50 @@
+To deploy a Solution in a MetalK8s cluster, an utility script is provided.
+This section describe, step by step, how to deploy a Solution using this
+tool.
+This script is located at the root of Metalk8s archive:
```
```suggestion
tool, located at the root of a MetalK8s archive:
```

alexandre-allard replied:
done

gdemonet commented on docs/operation/solutions.rst:
```diff
@@ -0,0 +1,50 @@
+Import Solution
+***************
+
+First, the Solution must be imported in MetalK8s cluster:
```
```suggestion
First, the Solution must be imported in the cluster (make the container
images available through the cluster registry):
```

alexandre-allard replied:
done

gdemonet commented on docs/operation/solutions.rst:
```diff
@@ -0,0 +1,50 @@
+Activate Solution
+*****************
+
+Then, the Solution must be activated:
```
```suggestion
Only one version of a Solution can be active at any point in time.
An active Solution version provides the cluster-wide resources to all other
versions of this Solution. To activate a version, run:
```

alexandre-allard replied:
done

gdemonet commented on docs/operation/solutions.rst:
```diff
@@ -0,0 +1,50 @@
+    ./solutions.sh import --archive </path/to/solution.iso>
+
+
+Activate Solution
```
```suggestion
Activate a Solution Version
```

alexandre-allard replied:
done

gdemonet commented on docs/operation/solutions.rst:
```diff
@@ -0,0 +1,50 @@
+Environment Creation
+********************
+
+The Solution will run inside an Environment, to create this environment, use:
```
```suggestion
Solutions are meant to be deployed in isolated namespaces, which we call
:term:`Environments <Environment>`. To create an Environment, run:
```

alexandre-allard replied:
done

gdemonet commented on docs/operation/solutions.rst:
```diff
@@ -0,0 +1,50 @@
+
+    ./solutions.sh create-env --name <environment-name>
+
+Adding Solution to an Environment
```
```suggestion
Adding a Solution Version to an Environment
```

alexandre-allard replied:
done

gdemonet commented on docs/operation/solutions.rst:
```diff
@@ -0,0 +1,50 @@
+Adding Solution to an Environment
+******************************
+
+The Solution must then be added to this Environment:
```
```suggestion
Solutions are packaged with an :term:`Operator`, and optionally an
associated web UI, to provide all required domain-specific logic.
To deploy a Solution Operator and its UI in an Environment, run:
```

alexandre-allard replied:
done

gdemonet commented on docs/operation/solutions.rst:
```diff
@@ -0,0 +1,50 @@
+  .. code::
+
+    ./solutions.sh add-solution --name <environment-name> \
+      --solution <solution-name> --version <solution-version>
```
After this command, maybe add a really short section to explain that further operations required to deploy and configure the Solution application itself (e.g. a Zenko cluster), our reader should refer to said Solution's documentation and interact with the Operator or UI we just deployed.

alexandre-allard replied:
done

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @gdemonet

bert-e commented:
# History mismatch



Merge commit #4a654fe1d581c96d7528fc45c63a36538bcb903a on the integration branch
`w/2.5/documentation/2309-enhance-solutions-doc` is merging a branch which is neither the current
branch `documentation/2309-enhance-solutions-doc` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `documentation/2309-enhance-solutions-doc` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

slaperche-scality commented on docs/operation/solutions.rst:
```diff
@@ -0,0 +1,62 @@
+
+  .. code::
+
+    /srv/scality/metalk8s-X.X.X/solutions.sh
```
I think in Sphinx we can insert the right version number instead of `X.X.X`

alexandre-allard replied:
done

slaperche-scality commented on docs/operation/solutions.rst:
```diff
@@ -0,0 +1,62 @@
+
+  .. code::
+
+    ./solutions.sh import --archive </path/to/solution.iso>
```
Absolute path? Or relative are also OK?

With the previous tooling I think absolute path was required.

alexandre-allard replied:
Both are Ok, yes previous script was only handling absolute paths.

slaperche-scality commented on docs/operation/solutions.rst:
```diff
@@ -0,0 +1,62 @@
+***************************
+
+Only one version of a Solution can be active at any point in time.
+An active Solution version provides the cluster-wide resources to all other
```
`the cluster-wide resources`, for example?

alexandre-allard replied:
CRDs

slaperche-scality replied:
Maybe mention it, i.e: `…provides the cluster-wide resources, such as CRDs, to all other…`

alexandre-allard replied:
done

gdemonet commented on docs/operation/solutions.rst:
```diff
@@ -0,0 +1,62 @@
+
+To deploy a Solution in a MetalK8s cluster, a utility script is provided.
+This section describes, step by step, how to deploy a Solution using this
+tool, located at the root of Metalk8s archive:
```
```suggestion
tool, located at the root of MetalK8s archive:
```

alexandre-allard replied:
done

gdemonet commented on docs/operation/solutions.rst:
```diff
@@ -0,0 +1,62 @@
+
+  .. code::
+
+    /srv/scality/metalk8s-X.X.X/solutions.sh
```
As @slaperche-scality suggested:

```suggestion
  .. parsed-literal::

    /srv/scality/metalk8s-|release|/solutions.sh
```

alexandre-allard replied:
Wouw, wonderful !

alexandre-allard replied:
done

alexandre-allard committed:
docs: add an upgrade/downgrade section for solutions

Add an explanation on how to do an upgrade/downgrade
of a deployed Solution in the developer guide.

Refs: #2309

alexandre-allard commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `documentation/2309-enhance-solutions-doc` into
`development/2.4`
* [w/2.5/documentation/2309-enhance-solutions-doc](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/documentation/2309-enhance-solutions-doc) will be merged into `development/2.5` 
* [w/2.6/documentation/2309-enhance-solutions-doc](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/documentation/2309-enhance-solutions-doc) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort





The following reviewers are expecting changes from the author, or must review again:

* @gdemonet

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

alexandre-allard commented:
/approve

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/caa4583472cb9835918de8b5da9bbf659e7744ea) did not succeed in branch w/2.6/documentation/2309-enhance-solutions-doc.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`

* :heavy_check_mark: `development/2.6`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`

* :heavy_check_mark: `development/2.6`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



Please check the status of the associated issue None.

Goodbye alexandre-allard-scality.
